lib/super_digit.rb

```diff
@@ -1,15 +1,31 @@
 # Superdigit
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(n)
+# Space Complexity - O(n^2)?
 def super_digit(n)
-   
-end
+  if n < 10
+    return n
+  end
+  
+  string_num = n.to_s
+  sum = 0
+  
+  string_num.each_char do |i|
+    sum += i.to_i
+  end
   
+  super_digit(sum)
+end
+
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(n); I'm not totally sure the time complexity of to_s or to_i, but 
+#   I think at worst it would be O(n) and since they run in sequence I think this maintains
+#   O(n).
+# Space Complexity - O(n^2); I'm thinking n^2 because every time super_digit is called,
+#   it creates a new variable string_num, whose size is relative to the size of n. So,  
+#   n multiplied n times.
 def refined_super_digit(n, k)
```

Comment:
This is the brute-force solution.  However you can find the super_digit of n and then use that k times.